lib/bakeAmbientOcclusion.js

```diff
@@ -43,20 +48,323 @@ function bakeAmbientOcclusion(gltf, options) {
+
+    // Clone materials, techniques, programs, and shaders as needed by primitives in other scenes.
+    for (var otherSceneID in gltf.scenes) {
+        if (gltf.scenes.hasOwnProperty(otherSceneID)) {
+            var otherScene = gltf.scenes[otherSceneID];
+            NodeHelpers.forEachPrimitiveInScene(gltf, otherScene, cloneShadingChain, parameters);
+        }
+    }
+
+    // Edit the shaders.
+    // This can mean tracking down the unlit diffuse color input from the technique.
+    addAoToShaders(gltf, Object.keys(parameters.techniqueCloneIDs), options);
+
+    // Edit the programs: add the ao attribute
+    for (var programID in parameters.programCloneIDs) {
+        if (parameters.programCloneIDs.hasOwnProperty(programID)) {
+            gltf.programs[programID].attributes.push('a_ambientOcclusion');
+        }
+    }
+
+    // Edit the techniques: add ao to attributes, add ao to parameters
+    for (var techniqueID in parameters.techniqueCloneIDs) {
+        if (parameters.techniqueCloneIDs.hasOwnProperty(techniqueID)) {
+            var technique = gltf.techniques[techniqueID];
+            technique.attributes.a_ambientOcclusion = 'vertex_ao';
+            technique.parameters.vertex_ao = {
+                semantic: 'VERTEX_AO',
+                type: WebGLConstants.FLOAT
+            };
+        }
+    }
+}
+
+// Helper for addVertexData
+function concatenateAoBuffers(primitive, meshPrimitiveID, parameters) {
+    if (parameters.aoBufferByPrimitive.hasOwnProperty(meshPrimitiveID)) {
+        var aoBuffer = parameters.aoBufferByPrimitive[meshPrimitiveID];
+        var vertexCount = aoBuffer.samples.length;
+        var samples = aoBuffer.samples;
+        var counts = aoBuffer.count;
+        for (var i = 0; i < vertexCount; i++) {
+            samples[i] = 1.0 - (samples[i]/counts[i]);
+        }
+        parameters.allAOData = parameters.allAOData.concat(samples);
+        parameters.primitiveOrder.push(primitive);
+        parameters.primitiveNames.push(meshPrimitiveID);
+        parameters.primitiveVertexCounts.push(vertexCount);
+    }
+}
+
+function addVertexData(gltf, scene, raytracerScene) {
+    // Get all the ao vertex data together, ordered parallel with the vertex data
+    // - append all the aoBuffers from the primitives
+    // - record the order for the accessors
+    var parameters = {
+        aoBufferByPrimitive: raytracerScene.aoBufferByPrimitive,
+        primitiveOrder: [],
+        primitiveNames: [],
+        primitiveVertexCounts: [],
+        allAOData: []
+    };
+
+    NodeHelpers.forEachPrimitiveInScene(gltf, scene, concatenateAoBuffers, parameters);
+
+    var primitiveOrder = parameters.primitiveOrder;
+    var allAOData = parameters.allAOData;
+    var allAODataLength = allAOData.length;
+
+    gltf.buffers.aoBuffer = {
+        input: {
+            byteLength : allAODataLength * 4, // (FLOAT). TODO: add option to bake to shorts
```

Comment:
Is quantization better than using shorts since we know the tight min/max (instead of `0...1`)?  If so, then we won't need shorts at all.